### PR TITLE
plan, engine: get rid of trim sorts optimizer

### PR DESCRIPTION
**Pull request by MichaHoffmann**
This is not really an optimizer in a way since it changes the meaning of the query. Its more an engine feature that we can do this since the engine itself does the sorting when presenting results. Every optimizer can expect that; thats why it makes sense just to get rid of sorts before even doing optimizations.

### CODE REVIEW

**Review by fpetkovski — APPROVED** [review]
Nice 👍

File: logicalplan/plan.go [hunk 1]
```diff
@@ -46,6 +46,9 @@ func New(expr parser.Expr, opts *query.Options) Plan {
 	setOffsetForAtModifier(opts.Start.UnixMilli(), expr)
 	setOffsetForInnerSubqueries(expr, opts)
 
+	// the engine handles sorting at the presentation layer
+	expr = trimSorts(expr)
```
Comment (yeya24) [inline logicalplan/plan.go:50]: How do we support sort_desc right now?
The presentation layer needs to know it needs to sort in reverse order?

Comment (yeya24) [reply logicalplan/plan.go:50]: And I wonder the behavior of topk and bottomk together with this change

Comment (MichaHoffmann) [reply logicalplan/plan.go:50]: Sorting is done here https://github.com/thanos-io/promql-engine/blob/7a183df0925bee2179aa351b235ed090b82dca9e/engine/engine.go#L475. Care needs to be taken to compute the sorting order before this step in the plan but if that is done then sorting functions are irrelevant for the remaining computation.

Comment (yeya24) [reply logicalplan/plan.go:50]: Thanks. Then this makes sense

**Review by yeya24 — APPROVED** [review]

**Pull request merged**